docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,356 @@
+Commit best practices
+---------------------
+
+How to split a change into commits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The golden rule to create good commits is to ensure that there is only one
+logical change per commit.
+
+This has several advantages amongst whose:
+
+- small commits are easier to review (a large pull request correctly divided
+  into logical commits is easier/faster to review than a medium-sized one with
+  less thought-out division)
+
+- logical commits are easier to revert (`e866b01f0553 <https://github.com/scality/metalk8s/commit/e866b01f05535925e80da20aca00417904422433>`_/`8208a170ac66 <https://github.com/scality/metalk8s/commit/8208a170ac66912ace018bcd00c058ad214d169b>`_)/cherry-pick
+  (`Pull request #1641 <https://github.com/scality/metalk8s/pull/1641>`_)
+
+- when looking for a regression (e.g. using ``git bisect``) it is easier to
+  find the root cause
+
+- make ``git log`` and ``git blame`` way more useful
+
+Examples
+^^^^^^^^
+
+Cosmetic changes
+::::::::::::::::
+
+Use a dedicated commit when you want to make cosmetic changes to the code
+(linting, whitespaces, alignment, renaming, …).
+
+Mixing cosmetics and functional changes is bad because the cosmetics (which
+tend to generate a lot of diff/noise) will obscure the important functional
+changes, making it harder to correctly determine whether the change is correct
+during the review.
```

Comment:
*Ideally, purely cosmetic changes which inflate the number of changes in a PR significantly, should go in a separate PR*